File: java-maven-spring-data-jdbc-dynamic/src/main/java/pl/exsio/querydsl/entityql/examples/spring_data_jdbc/entity/Book.java
package pl.exsio.querydsl.entityql.examples.spring_data_jdbc.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;

import java.math.BigDecimal;

public class Book {

    @Id
    @Column("BOOK_ID")
    private final Long id;
    private final String name;
    private final String desc;
    private final BigDecimal price;

    public Book(Long id, String name, String desc, BigDecimal price) {
        this.id = id;
        this.name = name;
        this.desc = desc;
        this.price = price;
    }

    @Override
    public String toString() {
        return "Book{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", desc='" + desc + '\'' +
                ", price=" + price +
                '}';
    }
}

File: kotlin-maven-spring-data-jdbc-dynamic/src/main/kotlin/pl/exsio/querydsl/entityql/examples/spring_data_jdbc/entity/Book.kt
package pl.exsio.querydsl.entityql.examples.spring_data_jdbc.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import java.math.BigDecimal

class Book(@Id @Column("BOOK_ID") var id: Long,
           var name: String,
           var desc: String?,
           var price: BigDecimal
)
